### PR TITLE
Add RecordingMicroAllocator to python interpreter

### PR DESCRIPTION
**Pull request by tanmaydas82**
BUG=http://b/246973049

### CODE REVIEW

**Review by paulinesho — APPROVED** [review]

File: tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py [hunk 1]
```diff
@@ -95,6 +95,17 @@ def from_bytes(self,
     return Interpreter(model_data, custom_op_registerers, arena_size,
                        num_resource_variables)
 
+  def print_allocations(self):
+    """Invoke the RecordingMicroAllocator to print the arena usage.
+
+    This should be called after `invoke()`.
+
+    Returns:
+      Status code of the C++ invoke function. A RuntimeError will be raised as
```
[the remark this answers is not in the document]
Comment (tanmaydas82) [reply tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py]: I overlooked it. Thank you for pointing this one! Changed the comment in the returns section.

**Review by paulinesho — APPROVED** [review]

File: tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py [hunk 2]
```diff
@@ -95,6 +95,17 @@ def from_bytes(self,
...
+  def print_allocations(self):
+    """Invoke the RecordingMicroAllocator to print the arena usage.
+
+    This should be called after `invoke()`.
+
+    Returns:
+      This method does not return anything, but It dumps the arena
+      usage to stderr.
+    """
+    return self._interpreter.PrintAllocations()
```
Comment (paulinesho) [inline tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py:107]: If this doesn't return anything, we shouldn't have `return` here, right?

**Review by paulinesho — CHANGES_REQUESTED** [review]
See comment in `tflm_runtime.py`

File: tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py (hunk 2 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/python/interpreter/src/tflm_runtime.py:107]: Correct. Let me remove that.

**Review by paulinesho — APPROVED** [review]

**Pull request merged**